vault/request_handling.go

```diff
@@ -333,6 +347,13 @@ func (c *Core) handleLoginRequest(req *logical.Request) (*logical.Response, *log
 
 		te.Policies = policyutil.SanitizePolicies(te.Policies, true)
 
+		// Prevent internal policies from being assigned to tokens
+		for _, policy := range te.Policies {
+			if strutil.StrListContains(nonAssignablePolicies, policy) {
```

Comment:
Very good catch on this! This makes me think if we should place the `nonAssignablePolicies` inside `te.create`.